include/picongpu/plugins/binning/Axis.hpp

```diff
@@ -0,0 +1,317 @@
+/* Copyright 2023 Tapish Narwal
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/plugins/binning/DomainInfo.hpp"
+#include "picongpu/plugins/binning/UnitConversion.hpp"
+
+#include <cstdint>
+
+namespace picongpu
+{
+    namespace plugins::binning
+    {
+        namespace axis
+        {
+            /**
+             * @brief Holds the range in axis space over which the binning will be done
+             */
+            template<typename T_Data>
+            class Range
+            {
+            public:
+                T_Data min; ///< Minimum of binning range
```

Comment:
Please move comments always above the variable and use `/**` to start the doxygen comment.
Could you please add to the comment if the value is in PIC or SI units.

Reply:
Changed comment style and added info that it is in SI units